h/auth/policy.py

```diff
@@ -341,6 +197,103 @@ def forget(self, _request):  # pylint: disable=no-self-use
         return []
 
 
+@interface.implementer(interfaces.IAuthenticationPolicy)
+class AuthClientPolicy(IdentityBasedPolicy):
+    """
+    An authentication policy for registered AuthClients.
+
+    Auth clients must be registered with grant type `client_credentials` and
+    will need to perform basic HTTP auth with their username and password set
+    to their auth client id, and secret.
+
+    A client can also pass an `X-Forwarded-User` header with the userid to
+    act as user in their authority. This will create a `request.user` and will
+    look like a normal login. This user will have an additional
+    principal, `client:{client_id}@{authority}` which lets you tell it apart
+    from regular users.
+
+    To differentiate between request with a token authenticated user and
+    a request with a forwarded user, the latter has an additional
+    principal, `client:{client_id}@{authority}`.
```

Comment:
This might be the same information twice?

Reply:
Oops yeah. I was writing the one above to try and capture all the information in the previous description but shorter. I missed out deleting this part as I expressed it.